substrate/frame/staking-async/src/pallet/mod.rs

```diff
@@ -1356,6 +1544,15 @@ pub mod pallet {
 				T::SlashDeferDuration::get(),
 				T::BondingDuration::get(),
 			);
+
+			// Ensure MaxPruningWeight is reasonable (not zero)
+			let max_pruning_weight = T::MaxPruningWeight::get();
+			assert!(
+				!max_pruning_weight.ref_time().is_zero() &&
+					!max_pruning_weight.proof_size().is_zero(),
+				"MaxPruningWeight must have non-zero ref_time and proof_size, got {:?}",
```

Comment:
what if it is an epsilon more than zero? it would make no progress if implemented correctly.

I think we should check that it is large enough to delete at least one item.